pootle/apps/accounts/utils.py

```diff
@@ -30,3 +34,177 @@ def get_user_by_email(email):
             return User.objects.get(email__iexact=email)
         except User.DoesNotExist:
             return None
+
+
+def merge_user(src_user, target_user):
+    """Merges one user to another.
+
+    The following are fields are updated (model: fields):
+    - units: submitted_by, commented_by, reviewed_by
+    - submissions: submitter
+    - suggestions: user, reviewer
+
+    :param src_user: `User` instance to merge from.
+    :param target_user: `User` instance to merge to.
+    """
+
+    # Move units to target_user.
+    for unit in src_user.submitted.iterator():
+        unit.submitted_by = target_user
+        unit.save()
+
+    for unit in src_user.commented.iterator():
+        unit.commented_by = target_user
+        unit.save()
+
+    for unit in src_user.reviewed.iterator():
+        unit.reviewed_by = target_user
+        unit.save()
+
+    # Move submissions to target_user.
+    for submission in src_user.submission_set.iterator():
+        submission.submitter = target_user
+
+        # Before we can save we first have to remove existing score_logs for
+        # this submission - they will be recreated on save with correct user.
+        for score_log in submission.scorelog_set.iterator():
+            score_log.delete()
+        submission.save()
+
+    # Move suggestions to target_user.
+    for suggestion in src_user.suggestions.iterator():
+        suggestion.user = target_user
+        suggestion.save()
+
+    for suggestion in src_user.reviews.iterator():
+        suggestion.reviewer = target_user
+        suggestion.save()
+
+
+def remove_units_created_by(user):
+    """Remove units created by user that have not had further
+    activity.
+
+    :param user: `User` to remove units for.
+    """
+    # Delete units created by user without submissions by others.
+    created_by_user = (user.submission_set
+                       .filter(type=SubmissionTypes.UNIT_CREATE))
+    for submission in created_by_user.iterator():
+        # Find submissions by other users on this unit.
+        other_subs = submission.unit.submission_set.exclude(submitter=user)
+
+        if not other_subs.exists():
+            submission.unit.delete()
+
+
+def revert_units_edited_by(user):
+    """Revert unit edits made by a user to previous edit.
+
+    :param user: `User` to revert submissions for.
+    """
+    editing_types = (SubmissionTypes.NORMAL,
+                     SubmissionTypes.SYSTEM,
+                     SubmissionTypes.UNIT_CREATE,
+                     SubmissionTypes.UPLOAD)
+
+    # Revert unit target where user is the last submitter.
+    for unit in user.submitted.iterator():
+
+        # Find the last submission by different user that updated the
+        # unit.target.
+        # NB: not sure if this is the best way.
+        last_edit = (unit.submission_set
+                     .exclude(submitter=user)
+                     .exclude(new_value__isnull=True)
+                     .exclude(field=SubmissionFields.COMMENT)
```

Comment:
This can be error-prone if new fields are added in the future that should be excluded too. What about having a `.filter(field__in=[SubmissionFields.TRANSLATION_FIELDS])`, where `SubmissionFields.TRANSLATION_FIELDS = [<list of fields to consider>]`?